admin-dev/themes/new-theme/js/pages/categories/name-to-link-rewrite-copier.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+/**
+ * Copies name of category to link rewrite input.
+ */
+export default class NameToLinkRewriteCopier {
+  constructor() {
+    ['category', 'root_category'].forEach((categoryType) => {
+      const $categoryForm = $('form[name="' + categoryType + '"]');
+
+      if (0 ===  $categoryForm.length) {
+        return;
+      }
+
+      $categoryForm.on('input', 'input[name^="' + categoryType + '[name]"]', (event) => {
+        const $nameInput = $(event.currentTarget);
+        const langId = $nameInput.closest('.js-locale-input').data('lang-id');
+
+        $categoryForm
+          .find('input[name="' + categoryType + '[link_rewrite][' + langId + ']"]')
+          .val(str2url($nameInput.val(), 'UTF-8'));
```

Comment:
What is `str2url`? Idon't see any import

Reply:
it comes from `ps-root/js/admin.js` which is included in every BO page, so its a global function. i know its not a clean way, but is there a better way to deal with it? :thinking: 

i saw this function was used the same way in other new JS file. :/

Reply:
Could you have the time to add it? in `admin-dev/themes/new-theme/js/components` ?